databases/foundationdb/Portfile

```diff
@@ -0,0 +1,92 @@
+# -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
+
+PortSystem          1.0
+PortGroup           github 1.0
+PortGroup           cmake 1.1
+
+name                foundationdb
+version             6.3.12
+categories          databases
+license             Apache-2
+maintainers         {me.com:spam_brian @brianwells} openmaintainer
+platforms           darwin
+description         FoundationDB is a distributed database providing ACID transactions.
+long_description    \
+    FoundationDB is a scalable, fault-tolerant, ordered \
+    key-value store with full ACID transactions.  This port \
+    will install version ${version} of the FoundationDB \
+    binaries suitable for development on macOS.
+
+homepage            https://www.foundationdb.org/
+
+github.setup        apple foundationdb ${version}
+
+checksums           rmd160  2bf9a258d5db140d232fb6355fc810010824c014 \
+                    sha256  034f69d4ddf7faccd7cae2a6438d64643ac98a3a9c7b3e6ed2a03105c480df95 \
+                    size    9694306
+
+depends_lib-append  port:mono
+
+cmake.generator     Ninja
+build.jobs          1
```

Comment:
Could you use here instead

`use_parallel_build no`

It would also be good to include a brief comment as to why its necessary to disable parallel builds.

Reply:
The FoundationDB build can easily run out of memory and crash, so they recommend using `ninja -j1` when building. It seems that using `use_parallel_build no` causes the option `-j${build.jobs}` to not be passed to Ninja, which is why I used `build.jobs 1`. I will add a comment in the Portfile to indicate why we want Ninja to build with only one job.

Reply:
OK that make sense, although we should investigate why setting `use_parallel_build no` doesn't do what it should with ninja, at least in this case.

Reply:
I'm guessing ninja defaults to enabling some level of parallelism, so just running `ninja` is not enough to limit things to one at a time (which it is with make for instance), hence `-j1` is needed....

Reply:
It looks like Ninja defaults to parallelism without needing `-j` to be passed: https://ninja-build.org/manual.html#_running_ninja

Reply:
Right. Sounds like this is something that then should be handled in base, or the cmake PG, so setting the usual option works as expected. But until then its fine you do it by hand here

Reply:
Hmmm, seems the cmake PG should already be handling this correctly

https://github.com/macports/macports-ports/blob/8a9bca9d57466ccc712f600b58d7b8d6cde79151/_resources/port1.0/group/cmake-1.1.tcl#L164

could you run a test, reverting to using `use_parallel_build no` and then run

```
sudo port -d configure foundationdb
```

and post the full log of what that gives ? I want to see what is generated for the configure / build commands...

Reply:
don't worry, I can replicate this myself with another port using cmake+ninja.... should be easy to address.

Reply:
Fixed with https://github.com/macports/macports-ports/commit/1db8f6311d91c3f9b9346de011e755a1c7887ce5

Can you please go back to just setting `use_parallel_build no`, thanks.

Reply:
Well, this is interesting. Digging deeper I find that both `build.jobs 1` and `use_parallel_build no` are resulting in `ninja all -j8 -v` being run... Fortunately I have 32 GB of memory on this iMac.

Reply:
there is also a bug in base, which I am about to submit a PR for. It assumes not passing any -jX flag when `use_parallel_build no` is set is the correct thing to do....

I think, with the fix in the cmake PG I just submitted (and it might take a while to get to you) the combination of

```
use_parallel_build no
build.jobs 1
```

should work...

Reply:
https://github.com/macports/macports-base/pull/236

Reply:
With both fixes in place I see: `ninja -j1 all -j8 -v` Perhaps the macports-base fix is sufficient? Not sure where the 8 jobs count is coming from and how `build.jobs 1` is overridden.

Reply:
No, those fixes are fine. They work just as I expect here. e.g. use another port that builds with cmake+ninja

with `use_parallel_build yes`

```
ninja -j8 all -j8 -v
```

with `use_parallel_build no`

```
ninja -j1 all -j1 -v 
```

The fact you are seeing `-j8 -v` at the end of the argument list suggests to me your aren't correctly running with the CMake PG fix.

Reply:
I figured it out.. It seems the `cmake.generator Ninja` line in the Portfile must come after the `use_parallel_build no` and/or `build.jobs 1` lines. Once I moved the `cmake.generator Ninja` line down things started working properly.